src/sinks/databend/service.rs

```diff
@@ -67,8 +66,12 @@ impl Finalizable for DatabendRequest {
 }
 
 impl MetaDescriptive for DatabendRequest {
-    fn get_metadata(&self) -> RequestMetadata {
-        self.metadata
+    fn get_metadata(&self) -> &RequestMetadata {
+        &self.metadata
+    }
+
+    fn take_metadata(&mut self) -> RequestMetadata {
+        std::mem::take(&mut self.metadata)
```

Comment:
These seem duplicated across the sinks - are we not able to centralize them?

Reply:
I don't think so, no. We can't provide a default impl for these trait methods because there's no supertrait it could depend on to know there's a self.metadata and be able to reference it.

Reply:
Yeah every struct that implements this trait will have a different `metadata` field to take from.